### PR TITLE
docs: start bringing structure from /docs to docusaurus

### LINKED ISSUE #1909: Bringing docs to the Backstage.io
## Plan
- [x] Move microsite to monorepo
- [x] Move structure from docs/README.md into sidebars.json and adjust headers in corresponding `.md` files 
```
// docs/overview/architecture-overview.md
---
id: architecture-overview
title: Architecture overview
---
```
```
// microsite/sidebars.json
 "Overview": [
     ...
      "overview/architecture-overview"
    ]
```

  - [x] Adjust Github Actions
  - [ ] Adjust DNS records for backstage.io

### PR DESCRIPTION
**Pull request by shmidt-i**
Progress towards https://github.com/spotify/backstage/issues/1909

### CODE REVIEW

**Review by Nek — APPROVED** [review]
👍

**Review by soapraj — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -4,99 +4,101 @@
 really 😆) you find broken links or missing content, please create an issue or,
 better yet, a pull request.
 
+# Plugins
+
 - Overview
   - [What is Backstage?](overview/what-is-backstage.md)
   - [Backstage architecture](overview/architecture-overview.md)
-  - [Architecture and terminology](overview/architecture-terminology.md)
+  - [Architect:ure and terminology](overview/architecture-terminology.md)
```
Comment (freben) [inline docs/README.md:12]: Oops

File: docs/features/software-templates/adding-templates.md [hunk 2]
```diff
@@ -15,7 +18,8 @@ metadata:
   # title of the template
   title: React SSR Template
   # a description of the template
-  description: Next.js application skeleton for creating isomorphic web applications.
+  description:
+    Next.js application skeleton for creating isomorphic web applications.
```
Comment (freben) [inline docs/features/software-templates/adding-templates.md:22]: Does this newline work? I think there needs to be a `|` or `>` after the colon on the previous line, no?

Comment (shmidt-i) [reply docs/features/software-templates/adding-templates.md:22]: I think this is prettier interference :) But good point, need to double check

**Review by soapraj — APPROVED** [review]

**Pull request merged**